examples/rust/components/blobby/src/lib.rs

```diff
@@ -0,0 +1,395 @@
+#![allow(clippy::missing_safety_doc)]
+// ^^^ This is for clippy complaining about something that wit bindgen is generating
+
+wit_bindgen::generate!({
+    world: "blobby",
+});
+
+use std::io::{Read, Write};
+
+use http::{
+    header::{ALLOW, CONTENT_LENGTH},
+    StatusCode,
+};
+
+use exports::wasi::http::incoming_handler::Guest;
+use wasi::blobstore::blobstore;
+use wasi::http::types::*;
+use wasi::logging::logging::{log, Level};
+use wrapper::OutputStreamWriter;
+
+mod wrapper;
+
+struct Error {
+    status_code: StatusCode,
+    message: String,
+}
+
+impl Error {
+    fn from_blobstore_error(e: blobstore::Error) -> Self {
+        Error {
+            status_code: StatusCode::BAD_GATEWAY,
+            message: format!("Error when communicating with blobstore: {}", e),
+        }
+    }
+
+    fn not_found() -> Self {
+        Error {
+            status_code: StatusCode::NOT_FOUND,
+            message: "Object not found".to_string(),
+        }
+    }
+}
+
+type Result<T> = std::result::Result<T, Error>;
+
+struct Blobby;
+
+const DEFAULT_CONTAINER_NAME: &str = "default";
+// TODO: Replace functionality for setting container name via header
+#[allow(dead_code)]
+const CONTAINER_HEADER_NAME: &str = "blobby-container";
+const CONTAINER_PARAM_NAME: &str = "container";
+
+/// A helper that will automatically create a container if it doesn't exist and returns an owned copy of the name for immediate use
+fn ensure_container(name: &String) -> Result<()> {
+    if !blobstore::container_exists(name).map_err(Error::from_blobstore_error)? {
+        blobstore::create_container(name).map_err(Error::from_blobstore_error)?;
+    }
+    Ok(())
+}
+
+fn send_response_error(response_out: ResponseOutparam, error: Error) {
+    let response = OutgoingResponse::new(Fields::new());
+    response
+        .set_status_code(error.status_code.as_u16())
+        .expect("Unable to set status code");
+    let response_body = response.body().unwrap();
```

Comment:
nit; `expect` explaining that this can only be called once would be more informational for an example (https://github.com/WebAssembly/wasi-http/blob/13dad387dc2cd5ba2849f531b1848d6c63a683d9/wit/types.wit#L277-L279)